subsys/net/ip/net_context.c

```diff
@@ -623,6 +623,61 @@ int net_context_listen(struct net_context *context, int backlog)
 	return -EOPNOTSUPP;
 }
 
+#if defined(CONFIG_NET_IPV4)
+struct net_pkt *net_context_create_ipv4(struct net_context *context,
+					struct net_pkt *pkt,
+					const struct in_addr *src,
+					const struct in_addr *dst)
+{
+	struct net_if_ipv4 *ipv4 = net_pkt_iface(pkt)->config.ip.ipv4;
+
+	NET_ASSERT(ipv4);
+	NET_ASSERT(((struct sockaddr_in_ptr *)&context->local)->sin_addr);
+
+	if (!src) {
+		src = ((struct sockaddr_in_ptr *)&context->local)->sin_addr;
+	}
+
+	if (net_is_ipv4_addr_unspecified(src)
+	    || net_is_ipv4_addr_mcast(src)) {
+		src = &ipv4->unicast[0].address.in_addr;
```

Comment:
We should change this to use the net_if_ipv4_select_src_addr() that was introduced recently.